packages/i18n/src/create-i18n.js

```diff
@@ -74,29 +74,29 @@ const I18N_HOOK_REGEXP = /^i18n\.(n?gettext|has_translation)(_|$)/;
  * Retrieve the domain to use when calling domain-specific filters.
  */
 /**
- * @typedef {(text: string, domain?: string) => string} __
+ * @typedef {<T extends string>(text: T, domain?: string) => import('./types').TranslatableText< T >} __
  *
  * Retrieve the translation of text.
  *
  * @see https://developer.wordpress.org/reference/functions/__/
  */
 /**
- * @typedef {(text: string, context: string, domain?: string) => string} _x
+ * @typedef {<T extends string>(text: T, context: string, domain?: string) => import('./types').TranslatableText< T >} _x
  *
  * Retrieve translated string with gettext context.
  *
  * @see https://developer.wordpress.org/reference/functions/_x/
  */
 /**
- * @typedef {(single: string, plural: string, number: number, domain?: string) => string} _n
+ * @typedef {<T extends string, P extends string>(single: T, plural: P, number: number, domain?: string) => import('./types').TranslatableText< T | P >} _n
```

Comment:
We could use descriptive type argument names more often than we do instead of using `T` for "generic type" practically everywhere.

This case jumps out because I usually thing that very generic `T` type is followed by `U`, but in this case `P` is chosen for the plural. In that case, `S` makes more sense for `single`'s type.

```suggestion
 * @typedef {<S extends string, P extends string>(single: S, plural: P, number: number, domain?: string) => import('./types').TranslatableText< S | P >} _n
```

But I'd really like if we used descriptive types argument names when we can. It seems straightforward when there's a 1:1 match:

```suggestion
 * @typedef {<Single extends string, Plural extends string>(single: Single, plural: Plural, number: number, domain?: string) => import('./types').TranslatableText< Single | Plural >} _n
```